src/java/htsjdk/samtools/cram/structure/ReadTag.java

```diff
@@ -374,10 +377,12 @@ public static Object readSingleValue(final byte tagType,
             case 'I':
                 final long val = byteBuffer.getInt() & 0xffffffffL;
                 if (val <= Integer.MAX_VALUE) {
-                    return (int) val;
+                    return (int)val;
                 }
-                throw new RuntimeException(
-                        "Tag value is too large to store as signed integer.");
+                SAMUtils.processValidationError(new SAMValidationError(SAMValidationError.Type.TAG_VALUE_TOO_LARGE,
+                        "Tag value " + val + " too large to store as signed integer.", null), validationStringency);
+                // convert to unsigned int stored in a long
+                return val;
```

Comment:
I realize that you were just trying to replicate the existing behavior here, but the existing behavior seems to simply be wrong (we can't handle "I" attributes properly in htsjdk). For consistency, I think `BinaryTagCodec.readSingleValue()` should also change to allow all values >= 0 and <= `BinaryTagCodec.MAX_UINT`